blobfuse/directoryapis.cpp

```diff
@@ -142,6 +158,22 @@ int azs_readdir(const char *path, void *buf, fuse_fill_dir_t filler, off_t, stru
                         stbuf.st_nlink = 1;
                         stbuf.st_size = listResults[i].content_length;
                         fillerResult = filler(buf, prev_token_str.c_str(), &stbuf, 0); // TODO: Add stat information.  Consider FUSE_FILL_DIR_PLUS.
+
+                        /// List attribute cache: cache the found file locally without the contents
+                        if(list_attribute_cache == true)
+                        {
+                            off_t length = listResults[i].content_length;
+                            int fd = open((mntPathString+prev_token_str).c_str(), O_RDWR | O_CREAT, S_IRWXU | S_IRWXG);
```

Comment:
do you need to get a lock for this file first?

Reply:
Hmm good catch. Alternatively I can use O_EXCL flag here, and this would ensure that if some other thread created the same file before this line does, this open will fail with EEXIST. (Explained here: http://man7.org/linux/man-pages/man2/open.2.html) - thoughts ? @AlexanderYukhanov @asorrin-msft

Reply:
may be, but what if some other will to create/delete it just after this line? i think if the introduced a synchronization mechanism - mutex + flock in one place, we need to follow it in other places. otherwise, it will be too difficult to reason about synchronization.

Reply:
Yes, I was just looking through this, we definitely need to grab the mutex lock.  This may also massively increase the size of the map, we may have to prioritize fixing that